ADAL/ADALAutomation/ADAutoMainViewController.m

```diff
@@ -44,158 +64,185 @@ - (void)didReceiveMemoryWarning {
     // Dispose of any resources that can be recreated.
 }
 
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
+{
+    (void)sender;
+    
+    if ([segue.identifier isEqualToString:@"showRequest"])
+    {
+        ADAutoRequestViewController *requestVC = segue.destinationViewController;
+        requestVC.completionBlock = sender[@"completionBlock"];
+    }
+    
+    
+    if ([segue.identifier isEqualToString:@"showResult"])
+    {
+        ADAutoResultViewController *resultVC = segue.destinationViewController;
+        resultVC.resultInfoString = sender[@"resultInfo"];
+        resultVC.resultLogsString = sender[@"resultLogs"];
+    }
+}
+
 
 - (IBAction)acquireToken:(id)sender
 {
-    ADAutoInputViewController* inputController = [ADAutoInputViewController new];
+    (void)sender;
     
-    [inputController startWithCompletionBlock:^(NSDictionary<NSString *,NSString *> *parameters)
-     {
-         if(parameters[@"error"])
+    ADAutoParamBlock completionBlock = ^void (NSDictionary<NSString *, NSString *> * parameters)
+    {
+        _resultLogs = [NSMutableString new];
+        
+        if(parameters[@"error"])
+        {
+            [self dismissViewControllerAnimated:NO completion:^{
+                [self displayResultJson:parameters[@"error"]];
+            }];
+            return;
+        }
+        
+        bool validateAuthority = YES;
+        if(parameters[@"validate_authority"])
```

Comment:
I mostly brought these over, and I generally agree with defines.

However, I think in this app is very much just for our automation testing, and thus, no need for retyping.
It is easier to just edit/view in place so it's more convenient for us.